WalletWasabi/Models/Label.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace WalletWasabi.Models
+{
+	public class Label : IEquatable<Label>
+	{
+		public static Label Empty { get; } = new Label();
+		public static char[] Separators { get; } = new[] { ',', ':' };
+		public IEnumerable<string> Labels { get; }
+		public bool IsEmpty { get; }
+
+		private string LabelString { get; }
+
+		public Label(params string[] labels) : this(labels as IEnumerable<string>)
+		{
+		}
+
+		public Label(IEnumerable<string> labels)
+		{
+			Labels = (labels
+				   ?.SelectMany(x => x?.Split(Separators, StringSplitOptions.RemoveEmptyEntries) ?? new string[0])
+				   ?.Select(x => x.Trim())
```

Comment:
All these `?.` are unnecessary because Linq extensions return `Enumerable` or similar, Microsoft methods do not return null, we shouldn't neither btw.

Reply:
It does matter:

![image](https://user-images.githubusercontent.com/9156103/64151367-e2c8ae00-ce32-11e9-845d-dc47311b123f.png)

![image](https://user-images.githubusercontent.com/9156103/64151432-0a1f7b00-ce33-11e9-883f-7dd226f066fb.png)

Reply:
Please remove all those `?.`, they are not needed and other will copy that pattern. Try something like this:

```c#
			labels = labels ?? Enumerable.Empty<string>();
			Labels = (labels
				   .SelectMany(x => x?.Split(Separators, StringSplitOptions.RemoveEmptyEntries) ?? new string[0])
				   .Select(x => x.Trim())
				   .Where(x => x != "")
				   .Distinct(StringComparer.OrdinalIgnoreCase)
				   .OrderBy(x => x)
				   ?? Enumerable.Empty<string>())
				   .ToArray();
```

Reply:
Good idea. Done.